srtcore/group.cpp

```diff
@@ -3348,19 +3399,34 @@ void CUDTGroup::send_CheckPendingSockets(const vector<SRTSOCKET>& pending, vecto
         }
         else
         {
+            int swait_timeout = 0;
+
+            // There's also a hidden condition here that is the upper if condition.
+            is_pending_blocked = (nsuccessful == 0);
+
+            // If this is the case when 
+            if (m_bSynSending && is_pending_blocked)
+            {
+                HLOGC(gslog.Debug, log << "grp/sendBroadcast: will block for " << m_iSndTimeOut << " - waiting for any writable in blocking mode");
+                swait_timeout = m_iSndTimeOut;
+            }
```

Comment:
Code duplication: the same was used in send_broadcast... Consider some common funtion.

Reply:
I will. There's a PR for it, but we need to be clear off with all potential problems to solve first. There are marked places that should make this process easy enough. Note however that the things happen differently in broadcast and backup, which means that even conditions for getting sockets blocked are not exactly the same.